fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt

```diff
@@ -129,6 +130,28 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
         requirePreference<Preference>(R.string.pref_key_custom_sponsored_stories_parameters).apply {
             isVisible = Config.channel.isNightlyOrDebug
         }
+
+        setupRemoteSettingsServer()
+    }
+
+    private fun setupRemoteSettingsServer() {
+        val prodPreference = requirePreference<RadioButtonPreference>(R.string.pref_key_server_prod).apply {
+            isVisible = Config.channel.isNightlyOrDebug
+        }
+        prodPreference.onClickListener {
+        }
+
+        val stagePreference = requirePreference<RadioButtonPreference>(R.string.pref_key_server_stage).apply {
+            isVisible = Config.channel.isNightlyOrDebug
+        }
+        stagePreference.onClickListener {
+        }
+
+        val serverUrlPreference = requireContext().settings().remoteSettingsServer
+        prodPreference.setCheckedWithoutClickListener(serverUrlPreference)
+        stagePreference.setCheckedWithoutClickListener(!serverUrlPreference)
+
+        addToRadioGroup(prodPreference, stagePreference)
```

Comment:
Since the preference is a boolean and there's just a binary choice between two radio buttons, would making this a switch preference simplify all of this? 

"Use Production server (staging will be used when disabled)"
- On - production
- Off - staging

Reply:
Yes that makes sense.